src/lib/task_listener.js

```diff
@@ -196,50 +189,47 @@ export default class TaskListener extends EventEmitter {
   }
 
   scheduleTaskPoll(nextPoll=this.taskPollInterval) {
-    this.pollTimeoutId = setTimeout(() => {
-      async () => {
-        clearTimeout(this.pollTimeoutId);
+    this.pollTimeoutId = setTimeout(async () => {
+      clearTimeout(this.pollTimeoutId);
 
-        try {
-          await this.getTasks();
-        }
-        catch (e) {
-          this.runtime.log('[alert-operator] task retrieval error', {
-              message: e.toString(),
-              err: e,
-              stack: e.stack
+      try {
+        await this.getTasks();
+      } catch (e) {
+        this.runtime.log('[alert-operator] task retrieval error', {
+            message: e.toString(),
+            err: e,
+            stack: e.stack
+        });
+      }
+      let stats = {
+        uptime: this.host.billingCycleUptime(),
+        interval: this.runtime.billingCycleInterval
+      };
+      let remainder = stats.interval - (stats.uptime % stats.interval);
+      if (remainder * this.runtime.taskQueue.slowdownDivisor < stats.interval) {
+        //slow down the polling because it's nearing the end of the billing cycle
+        if (this.taskPollInterval === this.runtime.taskQueue.pollInterval) {
+          this.taskPollInterval = this.runtime.taskQueue.pollInterval * this.runtime.taskQueue.pollIntervalMultiplier;
+
+          this.runtime.log('polling', {
+            message: 'polling interval adjusted',
+            oldInterval: this.runtime.taskQueue.pollInterval,
+            newInterval: this.runtime.taskQueue.pollInterval * this.runtime.taskQueue.pollIntervalMultiplier
```

Comment:
Lots of references to `this.runtime.taskQueue`, I think we should cache it in a local variable, say, `tq` to improve readability.